uber/models/__init__.py

```diff
@@ -846,9 +870,25 @@ def lookup_promo_code(self, code):
             except Exception:
                 pass
             else:
-                clause = clause.or_(PromoCode.id == promo_code_id)
+                clause = clause.or_(model.id == promo_code_id)
+
+            return self.query(model).filter(clause).order_by(model.normalized_code.desc()).first()
+
+        def create_promo_code_group(self, attendee, name, badges):
```

Comment:
I agree, I guess my issue with doing this is that we end up putting a lot of class-specific functions in the Session class, which is A) kinda hard to find if you don't know we're doing that and 2) going to clog up the Session class with a lot of sorta-random stuff. Like, logically speaking these functions don't deal with the session so much as they deal with PromoCode and PromoCodeGroup's relationship _to_ the session. I don't know how we'd organize the code to reflect that.

Reply:
Yeah, that's a fair point.  If we wanted to, we could add another decorator to transplant these things into ``SessionMixin``, e.g. in ``attendee.py`` we could have something like

```
@session_monkeypatch
class SessionMixin:
    def create_promo_code_group(self, attendee, name, badges):
        ...
```

so that instead of having one giant ``SessionMixin`` class we instead have a bunch of local ones located in whichever files have the associated models.

Definitely out of scope for this ticket, but something to keep in mind if you ever feel like ``SessionMixin`` needs to be broken up.